### PR TITLE
Refactor chain events into separate repo.

### PR DESCRIPTION
**Pull request by jnaviask**
## Description
Removes all event logic from `shared/` and relocates it to https://github.com/hicommonwealth/chain-events, which is published to an npm package: `commonwealth-chain-events`.

## Motivation and Context
This is the first step toward fixing #455. By relocating the code to another project, we open the possibility of further separating the functionality, e.g. adding a queue to bridge chain events with the server-side event handlers.

## How has this been tested?
Got it to build, ensured chain event proposals load OK. Ensured that webpack chunking was not broken on non-chain communities. Commenting on proposals works. Notifications not tested.

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [x] no

### CODE REVIEW

**Comment by drewstone** [comment]
Would be great to:
- integrate offence PR after review
- then break this out and coordinate with Seedcode on where to edit events.

File: client/scripts/controllers/chain/edgeware/signaling.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { ProposalModule } from 'models';
 import SubstrateChain from 'controllers/chain/substrate/shared';
 import SubstrateAccounts, { SubstrateAccount } from 'controllers/chain/substrate/account';
 import { SubstrateCoin } from 'adapters/chain/substrate/types';
-import { SubstrateEntityKind } from 'events/substrate/types';
+import { SubstrateEntityKind } from '@commonwealth/chain-events/dist/src/substrate/types';
```
Comment (drewstone) [inline client/scripts/controllers/chain/edgeware/signaling.ts:10]: why do we have `dist` here? Can it not import TS directly like we are doing with the API types?

Comment (jnaviask) [reply client/scripts/controllers/chain/edgeware/signaling.ts:10]: I can try messing with the exports / package.json config to fix that, we may need a top level index.ts file with all the imports together

Comment (jnaviask) [reply client/scripts/controllers/chain/edgeware/signaling.ts:10]: but yeah -- there is "no such thing" as a fully TS npm module. you have to compile it to .d.ts/.js pairs of types/implementation. this is just how the system works :\

**Comment by drewstone** [comment]
Last note would be to ensure that tests are working on the repo. If the tests failing are event tests we should move those out.

**Pull request merged**